sw/device/lib/dif/dif_entropy_src.c

```diff
@@ -133,11 +167,84 @@ dif_entropy_src_result_t dif_entropy_src_read(const dif_entropy_src_t *entropy,
   return kDifEntropySrcOk;
 }
 
+dif_entropy_src_result_t dif_entropy_src_fifo_read(
+    const dif_entropy_src_t *entropy, uint32_t *buf, size_t len) {
+  if (entropy == NULL) {
+    return kDifEntropySrcBadArg;
+  }
+
+  uint32_t reg = mmio_region_read32(entropy->params.base_addr,
+                                    ENTROPY_SRC_FW_OV_CONTROL_REG_OFFSET);
+  if (bitfield_field32_read(reg, ENTROPY_SRC_FW_OV_CONTROL_FW_OV_MODE_FIELD) !=
+      0xa) {
+    return kDifEntropySrcInvalidFwOverrideSettings;
+  }
+
+  reg = mmio_region_read32(entropy->params.base_addr,
+                           ENTROPY_SRC_OBSERVE_FIFO_THRESH_REG_OFFSET);
+  if (reg < len) {
+    return kDifEntropySrcInvalidFifoReadLen;
+  }
+
+  do {
+    reg = mmio_region_read32(entropy->params.base_addr,
+                             ENTROPY_SRC_INTR_STATE_REG_OFFSET);
+  } while (!bitfield_bit32_read(
+      reg, ENTROPY_SRC_INTR_STATE_ES_OBSERVE_FIFO_READY_BIT));
+
+  for (size_t i = 0; i < len; ++i) {
+    reg = mmio_region_read32(entropy->params.base_addr,
+                             ENTROPY_SRC_FW_OV_RD_DATA_REG_OFFSET);
+    if (buf != NULL) {
+      buf[i] = reg;
+    }
+  }
+
+  // Clear the status bit.
+  reg = mmio_region_read32(entropy->params.base_addr,
+                           ENTROPY_SRC_INTR_STATE_REG_OFFSET);
+  reg = bitfield_bit32_write(
+      reg, ENTROPY_SRC_INTR_STATE_ES_OBSERVE_FIFO_READY_BIT, true);
+  mmio_region_write32(entropy->params.base_addr,
+                      ENTROPY_SRC_INTR_STATE_REG_OFFSET, reg);
```

Comment:
This is supposed to clear only the `es_observe_fifo_ready` bit, right?
```suggestion
  reg = bitfield_bit32_write(
      0, ENTROPY_SRC_INTR_STATE_ES_OBSERVE_FIFO_READY_BIT, true);
  mmio_region_write32(entropy->params.base_addr,
                      ENTROPY_SRC_INTR_STATE_REG_OFFSET, reg);
```